Sources/StreamChat/WebSocketClient/WebSocketClient.swift

```diff
@@ -135,23 +139,18 @@ class WebSocketClient {
         source: WebSocketConnectionState.DisconnectionSource = .userInitiated,
         completion: @escaping () -> Void
     ) {
-        connectionState = .disconnecting(source: source)
-        engineQueue.async { [engine, eventsBatcher] in
-            engine?.disconnect()
-
-            eventsBatcher.processImmediately(completion: completion)
+        switch connectionState {
+        case .initialized, .disconnected, .disconnecting:
+            connectionState = .disconnected(source: source)
+        case .connecting, .waitingForConnectionId, .connected:
+            connectionState = .disconnecting(source: source)
```

Comment:
when disconnecting, if we are already disconnected, we should not put the state to `disconnecting` since the `engine` won't do anything, and so it won't report the `disconnected` afterwards. For this reason, we need to instantly report as `disconnected` for this scenario.